src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyAttributes/HubProxyAttributes.csproj

```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Any reason we feel the need to have a separate assembly? Could we put this in SignalR.Client? I ask because the assembly will end up in users build outputs since it is referenced by app code.

Reply:
Yes, called it out in the summary
> * Put the attributes in the Client.Core package
>   - Only projects targeting the new client package will get the attributes
>   - Attribute can be placed on valid methods but wouldn't do anything + compile error if source generator isn't also referenced

Reply:
I guess further question - any reason we couldn't put the source generator in the Client.Core package? If it's only triggered by the presence of these attributes, we're good no?

Reply:
Theoretically we could? But then you're limiting the generator to new clients which is an unnecessary restriction.

> I ask because the assembly will end up in users build outputs since it is referenced by app code.

Why do we care about a single new dll?

Reply:
I'm not really worried about the binary. I wanted to understand why we're choosing to ship the generator in a way that is markedly different from others. 

> But then you're limiting the generator to new clients which is an unnecessary restriction.

I feel that having it support arbitrary versions might be problematic if the code generation has to change. It also reduces the testing matrix dramatically every time you release a new version of this since you only ever have to test one version (vs n-versions). And if we aren't testing n-versions as I suspect, maybe it's a reason to tie to SignalR.Client package?

Reply:
I had an offline discussion about this. I'm still not super convinced that trying to support older versions of the client lib is really an important goal but if the engineering team feels very strongly about it, I'm not going to stand in their way. 

We should bring the assembly name and new attributes thru API review.